### PR TITLE
Add new package: quicksilver

### PR DESCRIPTION
**Pull request by darmac**

### CODE REVIEW

File: var/spack/repos/builtin/packages/quicksilver/package.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+from spack import *
+
+
+class Quicksilver(MakefilePackage):
+    """Quicksilver is a proxy application that represents some elements of
+    the Mercury workload by solving a simpli?ed dynamic monte carlo
+    particle transport problem.  Quicksilver attempts to replicate the
+    memory access patterns, communication patterns, and the branching or
+    divergence of Mercury for problems using multigroup cross sections."""
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/quicksilver/package.py:14]: Can you separate the docstring into a short first sentence and a longer description in another paragraph?

File: var/spack/repos/builtin/packages/quicksilver/package.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    homepage = "https://codesign.llnl.gov/quicksilver.php"
+    git      = "https://github.com/LLNL/Quicksilver.git"
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/quicksilver/package.py:17]: According to Quicksilver's Github repository:

> Instructions to build Quicksilver can be found in the Makefile. Quicksilver is a relatively easy to build code with no external dependencies (except MPI and OpenMP)

Should we add MPI as a dependency then? Maybe conditional on a variant?

**Review by alalazo — CHANGES_REQUESTED** [review]
A few request for changes.  Can you also tag this application as a `proxy-app`? You can check which others are tagged by looking at:
```console
$ spack list -t proxy-app
```

**Comment by alalazo** [comment]
Pinging @richards12 in case he's interested in giving a few more hints on how the package for "Quicksilver" should look like.

**Comment by junghans** [comment]
@alalazo I added mpi and openmp support in #16811.

**Comment by alalazo** [comment]
@darmac I went with @junghans version of the package since it was ready to be merged, hope you don't mind (and thanks for all your contributions!)

**Comment by darmac** [comment]
That's OK, I'll test it on my platform. : )

**Pull request closed without merging**